tests/core/test_db_conversion.py

```diff
@@ -66,65 +75,81 @@ async def test_blocks(self):
         with TempFile() as in_file, TempFile() as out_file:
 
             async with aiosqlite.connect(in_file) as conn:
+
+                await conn.execute("pragma journal_mode=OFF")
+                await conn.execute("pragma synchronous=OFF")
+                await conn.execute("pragma locking_mode=exclusive")
+
                 db_wrapper1 = DBWrapper(conn, 1)
                 block_store1 = await BlockStore.create(db_wrapper1)
-                coin_store1 = await CoinStore.create(db_wrapper1, 0)
-                hint_store1 = await HintStore.create(db_wrapper1)
-
-                for hint in hints:
-                    await hint_store1.add_hints([(hint[0], hint[1])])
+                coin_store1 = await CoinStore.create(db_wrapper1, uint32(0))
+                if with_hints:
+                    hint_store1 = await HintStore.create(db_wrapper1)
+                    for h in hints:
+                        await hint_store1.add_hints([(h[0], h[1])])
+                else:
+                    hint_store1 = None
 
                 bc = await Blockchain.create(
                     coin_store1, block_store1, test_constants, hint_store1, Path("."), reserved_cores=0
                 )
                 await db_wrapper1.commit_transaction()
 
                 for block in blocks:
-                    await _validate_and_add_block(bc, block)
+                    # await _validate_and_add_block(bc, block)
+                    results = PreValidationResult(None, uint64(1), None, False)
+                    result, err, _, _ = await bc.receive_block(block, results)
+                    assert err is None
 
-                # now, convert v1 in_file to v2 out_file
-                await convert_v1_to_v2(in_file, out_file)
+            # now, convert v1 in_file to v2 out_file
```

Comment:
from here and below, one level of indentation was removed to close the database connection to the in-table before running the conversion function (the function being tested). The conversion function opens the database in exclusive mode, so technically, it wasn't exclusive prior to this change.